Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXOReportSubmissionPolicy.Tests.ps1

```diff
@@ -0,0 +1,267 @@
+            }
+        }
+
+        Context -Name 'ReportSubmissionPolicy update needed.' -Fixture {
+            BeforeAll {
+                $testParams = @{
+                    Ensure                                 = 'Present'
+                    Credential                             = $Credential
+                    IsSingleInstance                       = 'Yes';
+                    Identity                               = "DefaultReportSubmissionPolicy"
+                    DisableQuarantineReportingOption       = $False;
+                    EnableCustomNotificationSender         = $False;
+                    EnableOrganizationBranding             = $False;
+                    EnableReportToMicrosoft                = $True;
+                    EnableThirdPartyAddress                = $False;
+                    EnableUserEmailNotification            = $False;
+                    PostSubmitMessageEnabled               = $True;
+                    PreSubmitMessageEnabled                = $True;
+                    ReportJunkToCustomizedAddress          = $False;
+                    ReportNotJunkToCustomizedAddress       = $False;
+                    ReportPhishToCustomizedAddress         = $False;
+                }
+
+                Mock -CommandName Get-ReportSubmissionPolicy -MockWith {
+                    return @{
+                        Ensure                                 = 'Present'
+                        Credential                             = $Credential
+                        IsSingleInstance                       = 'Yes';
+                        Identity                               = "DefaultReportSubmissionPolicy"
+                        DisableQuarantineReportingOption       = $False;
+                        EnableCustomNotificationSender         = $False;
+                        EnableOrganizationBranding             = $False;
+                        EnableReportToMicrosoft                = $False;
+                        EnableThirdPartyAddress                = $False;
+                        EnableUserEmailNotification            = $False;
+                        PostSubmitMessageEnabled               = $False;
+                        PreSubmitMessageEnabled                = $False;
+                        ReportJunkToCustomizedAddress          = $False;
+                        ReportNotJunkToCustomizedAddress       = $False;
+                        ReportPhishToCustomizedAddress         = $False;
+                    }
+                }
+
+                Mock -CommandName Set-ReportSubmissionPolicy -MockWith {
+                    return @{
+
+                    }
+                }
+            }
+
+            It 'Should return false from the Test method' {
+                Test-TargetResource @testParams | Should -Be $false
+            }
+
+            It 'Should Successfully call the Set method' {
+                Set-TargetResource @testParams
```

Comment:
How should Pester validate that the function ran successfully. My guess is that a "Should -Invoke Set-ReportSubmissionPolicy" would be a good fit here. That checks if the Set-ReportSubmissionPolicy mock is called, which makes sense if the policy should be updated.

More info: https://pester.dev/docs/commands/Should#invoke